transports/daily/src/transport.ts

```diff
@@ -24,34 +24,106 @@ import {
   logger,
 } from "@pipecat-ai/client-js";
 
+import {MediaStreamRecorder} from "../../../lib/wavtools";
+
 export interface DailyTransportAuthBundle {
   room_url: string;
   token: string;
 }
 
 export interface DailyTransportConstructorOptions {
+  bufferLocalAudioUntilBotReady?: boolean;
   dailyFactoryOptions?: DailyFactoryOptions;
 }
 
+export enum DailyRTVIMessageType {
+  AUDIO_BUFFERING_STARTED = "audio-buffering-started",
+  AUDIO_BUFFERING_STOPPED = "audio-buffering-stopped",
+}
+
 export class DailyTransport extends Transport {
   private declare _daily: DailyCall;
   private _dailyFactoryOptions: DailyFactoryOptions;
+  private _bufferLocalAudioUntilBotReady: boolean;
   private _botId: string = "";
   private _selectedCam: MediaDeviceInfo | Record<string, never> = {};
   private _selectedMic: MediaDeviceInfo | Record<string, never> = {};
   private _selectedSpeaker: MediaDeviceInfo | Record<string, never> = {};
 
+  private static RECORDER_SAMPLE_RATE = 16_000
+  private static RECORDER_CHUNK_SIZE = 512
+  private currentAudioTrack: MediaStreamTrack | null = null;
+  private audioQueue: ArrayBuffer[] = [];
+  private declare mediaStreamRecorder: MediaStreamRecorder;
+
   constructor({
     dailyFactoryOptions = {},
+    bufferLocalAudioUntilBotReady = false,
   }: DailyTransportConstructorOptions = {}) {
     super();
     this._dailyFactoryOptions = dailyFactoryOptions;
+    this._bufferLocalAudioUntilBotReady = bufferLocalAudioUntilBotReady;
+  }
+
+  private setupRecorder(): void {
+    this.mediaStreamRecorder = new MediaStreamRecorder({ sampleRate: DailyTransport.RECORDER_SAMPLE_RATE });
```

Comment:
wrap at 80? do we have a linter/prettier set up for our pipecat js stuffs?

Reply:
Done.